### ISSUE TITLE
Altitude level above y=256 wraps around, inverting physics.

### ISSUE THREAD

**Issue by aramperes** [GlowstoneMC/Glowstone#337]
When you reach a level above y=256, physics are inverted; gravity goes upwards, so you float to a certain point. That point is actually defined by the inverse of the ground level.

I took the example of having the highest ground level of **70**.  If you teleport from the same position to the block limit, **256**, using `/tp ~ 256 ~`, you will not fall. Instead, you will "fall" upwards to level **326**. The reason for that limit is because **326-256=70**. So, you stop at that position because you are supposedly touching the ground.

Here's a video of that example: http://momoperes.ca/videos/velocity_bug.mp4

This does not seem to affect dropped items, but they seem very glitchy too. If you drop one under you while floating, you will be able to pick it up while it is 30 blocks under you.

**Comment by mastercoms** [comment]
Thanks for this analysis! I wasn't able to figure out what caused this bug, but now it makes a lot more sense!

Maybe this can be a feature :stuck_out_tongue: like skylands or something.

**Comment by aramperes** [comment]
Update:

I've done some more debugging on this issue. It seems that the but is related to the `world.getBlockAt(x,y,z)` method, which resets to 0 after passing the 256 mark. Take a look at this debug log, where "On type: Y" is the `block.getBlockY()` result, and "eye: Y" is the `getEyeLocation().getBlockY()`:

```
[INFO] On type: 254 / eye: 254
[INFO] On type: 255 / eye: 255
[INFO] On type: 0 / eye: 256
[INFO] On type: 1 / eye: 257
[INFO] On type: 2 / eye: 258
```

I'll look into what might be happening, but it's probably related to this (GlowWorld):

```
@Override
public GlowBlock getBlockAt(int x, int y, int z) {
    return new GlowBlock(getChunkAt(x >> 4, z >> 4), x, y & 0xff, z); //y & 0xff resets to 0 when reaching 256
}
```

**Edit. Replacing `y & 0xff` with `y` fixes everything.**

**Comment by aramperes** [comment]
Fixed in ac8b563.
